lib/Cron/Cleanup.php

```diff
@@ -57,28 +55,24 @@ public function __construct(ITimeFactory $time,
 	}
 
 	protected function run($argument) {
-		$this->logger->debug('Run cleanup job for text sessions');
-		$inactive = $this->sessionService->findAllInactive();
-		/** @var Session $session */
-		foreach ($inactive as $session) {
-			$activeSessions = $this->sessionService->getActiveSessions($session->getDocumentId());
-			if (count($activeSessions) > 0) {
+		$this->logger->debug('Run cleanup job for text documents');
+		$documents = $this->documentService->getAll();
+		foreach ($documents as $document) {
+			$allSessions = $this->sessionService->getAllSessions($document->getId());
```

Comment:
I'd think we could even do this for all documents with no *active* Sessions. But maybe we can also do that later.

Reply:
Thinking about this again, wouldn't this bring back the problem where the state is reset while a session is inactive but opens again later? However I think #3921 would be the better way of handling this case and then we could also cleanup more aggressively